src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -219,11 +222,14 @@ Copyright (c) .NET Foundation. All rights reserved.
       <TrimMode Condition="'$(TrimMode)' == '' And $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '6.0'))">copyused</TrimMode>
       <TrimMode Condition="'$(TrimMode)' == '' And $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0'))">partial</TrimMode>
       <TrimMode Condition="'$(TrimMode)' == ''">full</TrimMode>
+      <!-- Ignore unsupported TrimMode settings in 8.0+ -->
+      <TrimMode Condition="('$(TrimMode)' != 'partial' And '$(TrimMode)' != 'full') And $([MSBuild]::VersionGreaterThan('$(TargetFrameworkVersion)', '7.0'))">full</TrimMode>
+      <!-- Escape hatch: _TrimMode can be used to turn on the deprecated behaviors. -->
+      <TrimMode Condition="'$(_TrimMode)' != ''">$(_TrimMode)</TrimMode>
 
       <!-- Set TrimmerDefaultAction for compat in < 7.0 -->
       <_TrimmerDefaultAction Condition="$([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '7.0'))">$(TrimmerDefaultAction)</_TrimmerDefaultAction>
       <_TrimmerDefaultAction Condition="'$(_TrimmerDefaultAction)' == '' And $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '6.0'))">$(TrimMode)</_TrimmerDefaultAction>
-      <_TrimmerDefaultAction Condition="'$(_TrimmerDefaultAction)' == '' And $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0'))">copy</_TrimmerDefaultAction>
```

Comment:
I don't understand this part of the change. This is as if we're trying to intentionally break 6.0 apps with 8.0 SDK?